src/docfx/SubCommands/DocumentBuilderWrapper.cs

```diff
@@ -225,7 +228,112 @@ private static DocumentBuildParameters ConfigToParameter(BuildJsonConfig config,
+                    FileMappingParameters fileMappingParameters;
+                    if (result.TryGetValue(version, out fileMappingParameters))
+                    {
+                        if (fileMappingParameters.Content != null)
+                        {
+                            fileMappingParameters.Content.Add(item);
+                        }
+                        else
+                        {
+                            fileMappingParameters.Content = new FileMapping(item);
+                        }
+                    }
+                    else
+                    {
+                        result[version] = new FileMappingParameters
+                        {
+                            Content = new FileMapping(item)
+                        };
+                    }
+                }
+            }
+
+            if (overwrite != null)
+            {
+                foreach (var item in overwrite.Items)
+                {
+                    var version = item.VersionName ?? string.Empty;
+                    FileMappingParameters fileMappingParameters;
+                    if (result.TryGetValue(version, out fileMappingParameters))
+                    {
+                        if (fileMappingParameters.Overwrite != null)
+                        {
+                            fileMappingParameters.Overwrite.Add(item);
+                        }
+                        else
+                        {
+                            fileMappingParameters.Overwrite = new FileMapping(item);
+                        }
+                    }
+                    else
+                    {
+                        result[version] = new FileMappingParameters
+                        {
+                            Overwrite = new FileMapping(item)
+                        };
+                    }
+                }
+            }
+
+            if (resource != null)
+            {
+                foreach (var item in resource.Items)
+                {
+                    var version = item.VersionName ?? string.Empty;
+                    FileMappingParameters fileMappingParameters;
+                    if (result.TryGetValue(version, out fileMappingParameters))
+                    {
+                        if (fileMappingParameters.Resource != null)
+                        {
+                            fileMappingParameters.Resource.Add(item);
+                        }
+                        else
+                        {
+                            fileMappingParameters.Resource = new FileMapping(item);
+                        }
+                    }
+                    else
+                    {
+                        result[version] = new FileMappingParameters
+                        {
+                            Resource = new FileMapping(item)
+                        };
+                    }
+                }
+            }
```

Comment:
Refactored. Change FileMappingParameters to a dictionary, so the property can be converted to a key as a parameter.